tests/integration/telemetry/stackdriver/vm/vm_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb"
+	loggingpb "google.golang.org/genproto/googleapis/logging/v2"
+	monitoring "google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/protobuf/proto"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+)
+
+func TestVMTelemetry(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("traffic.reachability"). // use stackdriver vm telemetry feature
+		Run(func(ctx framework.TestContext) {
+			// Set up strict mTLS. This gives a bit more assurance the calls are actually going through envoy,
+			// and certs are set up correctly.
+			ctx.Config().ApplyYAMLOrFail(ctx, ns.Name(), `
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+spec:
+  mtls:
+    mode: STRICT
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: send-mtls
+spec:
+  host: "*.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+`)
+			ports := []echo.Port{
+				{
+					Name:     "http",
+					Protocol: protocol.HTTP,
+					// Due to a bug in WorkloadEntry, service port must equal target port for now
+					InstancePort: 8090,
+					ServicePort:  8090,
+				},
+			}
+
+			// builder to build the instances iteratively
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&clt, echo.Config{
+					Service:   "client",
+					Namespace: ns,
+					Ports:     ports,
+					Pilot:     p,
+					Subsets: []echo.SubsetConfig{
+						{
+							Annotations: map[echo.Annotation]*echo.AnnotationValue{
+								echo.SidecarBootstrapOverride: {
+									Value: sdBootstrapConfigMap,
+								},
+							},
+						},
+					},
+				}).
+				BuildOrFail(t)
+
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&srv, echo.Config{
+					Service:       "server",
+					Namespace:     ns,
+					Ports:         ports,
+					Pilot:         p,
+					DeployAsVM:    true,
+					VMImage:       "app_sidecar_ubuntu_bionic",
```

Comment:
nit: do you think it might be better if we use [`vm.DefaultVMImage`](https://github.com/istio/istio/blob/master/tests/integration/pilot/vm/util.go#L20) here? Just in case in the future the default image were to change?

Reply:
Yup. Makes sense to me.

Reply:
done.